List mode search endpoint

valya opened the pull request.

Commit:
:construction: super basic version of the search endpoint

This version is built using a hacked up version of the Google Drive
node. I need to properly create a v2 for it but it's does work.

Commit:
:construction: fixed up type errors and return urls

Commit:
:sparkles: add v3 of Google Drive node with RLC

Commit:
Merge remote-tracking branch 'origin/n8n-4387-resource-loader' into n8n-4370-implement-search-endpoints-be

Commit:
:sparkles: add RLC to Google Drive Shared Drive operations

BHesseldieck commented on packages/workflow/src/Interfaces.ts:
```diff
@@ -1087,6 +1105,13 @@ export interface INodeType {
+			[key: string]: (
+				this: ILoadOptionsFunctions,
+				filter?: string,
+				paginationToken?: unknown,
```
ref previous comment

Commit:
Merge remote-tracking branch 'origin/n8n-4387-resource-loader' into n8n-4370-implement-search-endpoints-be

Commit:
:recycle: address some small changes requested in review

Commit:
:bug: move list search out of /nodes/ and add check for required param

Commit:
:sparkles: google drive folder search

Commit:
:sparkles: google drive search sort by name

Commit:
:sparkles: add searchable flag for RLC

Commit:
:pencil2: fix google drive wording for v3

valya commented on packages/cli/src/requests.d.ts:
```diff
@@ -285,6 +285,25 @@ export type NodeParameterOptionsRequest = AuthenticatedRequest<
+		currentNodeParameters: string;
+		credentials: string;
+		filter?: string;
+		paginationToken?: string;
```
We talked it over and decided to keep it as `paginationToken` just for clarity.

valya commented on packages/cli/src/Server.ts:
```diff
@@ -932,6 +940,51 @@ class App {
+		// Returns parameter values which normally get loaded from an external API or
+		// get generated dynamically
+		this.app.get(
+			`/${this.restEndpoint}/node-list-search`,
```
I ended up reverting this change. It was causing weird issues since the community nodes endpoints are also under `/nodes/`.

valya committed:
Trello and Airtable search backend (#3974)

* :sparkles: add search to Trello boards

* :sparkles: add RLC to Trello cards

* :recycle: use new versioning system for Trello v2

* :bug: move list search out of /nodes/ and add check for required param

* :sparkles: re-add trello search methods

* :goal_net: throw error if RLC search isn't sent a method name

This will likely be removed when the declarative style of search has
been added.

* :sparkles: add requires filter field to RLC search

* :sparkles: add searchable flag to Trello searches

* :sparkles: add RLC for cardId and boardId on all operations

* :sparkles: add ID and URL RLC to Airtable
